overlord/snapstate/snapstate.go

```diff
@@ -554,7 +550,7 @@ func Set(s *state.State, name string, snapst *SnapState) {
 	if snaps == nil {
 		snaps = make(map[string]*json.RawMessage)
 	}
-	if snapst == nil || (len(snapst.Sequence) == 0 && snapst.Candidate == nil) {
+	if snapst == nil || (len(snapst.Sequence) == 0) {
```

Comment:
this means that a few of Set(st, name, snapst) will do nothing now, we need to be careful about what to do about that